syntax = "proto3";

package os_remote;


service OSRemote {
    rpc Init (ArgRequest) returns (ArgReply) {
    }
    rpc Open (ArgRequest) returns (ArgReply) {
    }
    rpc Delete (ArgRequest) returns (ArgReply) {
    }
    rpc Access (ArgRequest) returns (ArgReply) {
    }
    rpc FullPathname (ArgRequest) returns (ArgReply) {
    }
    rpc Randomness (ArgRequest) returns (ArgReply) {
    }
    rpc Sleep (ArgRequest) returns (ArgReply) {
    }
    rpc CurrentTime (ArgRequest) returns (ArgReply) {
    }
    rpc GetLastError (ArgRequest) returns (ArgReply) {
    }
    rpc CurrentTimeInt64 (ArgRequest) returns (ArgReply) {
    }
    rpc Write (ArgRequest) returns (ArgReply) {
    }
    rpc Read (ArgRequest) returns (ArgReply) {
    }
    rpc Truncate (ArgRequest) returns (ArgReply) {
    }
    rpc Sync (ArgRequest) returns (ArgReply) {
    }
    rpc FileSize (ArgRequest) returns (ArgReply) {
    }
    rpc FileControl (ArgRequest) returns (ArgReply) {
    }
    rpc SectorSize (ArgRequest) returns (ArgReply) {
    }
    rpc DeviceCharacteristics (ArgRequest) returns (ArgReply) {
    }
    rpc Close (ArgRequest) returns (ArgReply) {
    }
    rpc Lock (ArgRequest) returns (ArgReply) {
    }
    rpc Unlock (ArgRequest) returns (ArgReply) {
    }
    rpc CheckReservedLock (ArgRequest) returns (ArgReply) {
    }
    rpc Fetch (ArgRequest) returns (ArgReply) {
    }
    rpc Unfetch (ArgRequest) returns (ArgReply) {
    }
}

message ArgRequest {
    bytes inarg = 1;
    uint32 outlen = 2;
}
message ArgReply {
    bytes outarg = 1;
}
//message OpenRequest {
//  bytes inarg = 1;
//  uint32 outlen = 2;
//}
//
//message OpenReply {
//  bytes outarg = 1;
//}
//message DeleteRequest {
//  bytes inarg = 1;
//  uint32 outlen = 2;
//}
//
//message DeleteReply {
//  bytes outarg = 1;
//}
